### DISCUSSION TITLE
Accessing images and videos cross-platt

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#14481] [Q&A] [answered]
I want to develop kind of a backup app that uploads the images and videos from the local device and possibly deletes them locally. I am already able to query the images and videos. But I have trouble rendering them and accessing them using File.OpenRead() and File.Delete(). I get exceptions like this:

"System.UnauthorizedAccessException: Access to the path '/storage/emulated/0/DCIM/Camera/VID_20191115_082542.mp4' is denied"

I do not have a clue, why. I already have the READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE permissions. Why am I not allowed to get the raw data of the files?

I was able to convert the files to byte[] like this:

```
byte[] buffer;
 using (InputStream inputStream = new BufferedInputStream(Uno.UI.ContextHelper.Current.ContentResolver.OpenInputStream(mediaUri)))
 using(ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
 {
     byte[] bufferLoc = new byte[4096];
     int bytesRead;
     while ((bytesRead = inputStream.Read(bufferLoc)) != -1)
     {
         outputStream.Write(bufferLoc, 0, bytesRead);
     }

     buffer = outputStream.ToByteArray();
 }
```

But if the file is a hughe video file, this would lead to memory issues. And it still would not allow me to delete the file after processing.

Does anybody have an idea what could be wrong on my end? Thank you!

EDIT:
It works with this code: https://github.com/CrossGeeks/CustomGallerySample
I copied it to a fresh Uno-App (Net8) and it does not. Is it due to different SDK-versions or so? I cannot spot any difference, though.

**Comment by limeniye** [comment, marked as answer]
Hello, @TopperDEL!
Have you read the following doc?
https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/appmodel/permissions?view=net-maui-8.0&tabs=windows

The first thing I thought was that you are using mediaURI, but I see that it is not available for android.
This is just an opinion, but it is necessary to understand in more detail.

**Reply by TopperDEL** [reply]
What irritates me is that I have access to the files as expected when using the project linked above. It is an old Xamarin Forums app. I copied the Code over, check the permissions, have them granted but it still fails.

I'll check again with a fresh brain today. Maybe I Spot the difference now. I'll also take a look at the other permissions class from you.
Thank you!

**Reply by TopperDEL** [reply]
Hm, it is working now. It seems I mangled up something with the manifest and the Permission-Check.
